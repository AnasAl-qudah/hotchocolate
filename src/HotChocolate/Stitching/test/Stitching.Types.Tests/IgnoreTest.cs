using System.Threading.Tasks;
using HotChocolate.Language;
using HotChocolate.Language.Utilities;
using HotChocolate.Stitching.Types.Strategies;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Stitching.Types;

public class IgnoreTest
{
    [Fact]
    public async Task Test()
    {
        DocumentNode source = Utf8GraphQLParser.Parse(@"
            interface TestInterface @ignore {
              foo: [Test2!]
            }

            type Test implements TestInterface {
              id: String!
              foo: Test2!
            }

            type Test2 {
              test: Test!
            }

            type Test3 {
              foo: Test! @ignore
            }

            type Test4 {
              foo: TestInterface!
            }
",
            ParserOptions.NoLocation);

        var ignoreStrategy = new IgnoreStrategy();

        DocumentNode result = ignoreStrategy.Apply(source);

        var schema = result.Print();
        schema.MatchSnapshot();
    }
}
