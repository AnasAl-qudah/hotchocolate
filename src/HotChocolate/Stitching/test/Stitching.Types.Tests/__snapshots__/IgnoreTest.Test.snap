type Test {
  id: String!
  foo: Test2!
}

type Test2 {
  test: Test!
}

type Test4 {
  foo: TestInterface!
}
